[TrafficManager] Real User Metrics (RUM) fixes

allencal opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

allencal committed:
[Traffic Manager] Adding new RealtimeUserMetrics preview feature to Traffic Manager.

[Traffic Manager] Adding new RealtimeUserMetrics preview feature to Traffic Manager.

allencal committed:
[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

allencal committed:
[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

allencal committed:
[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

allencal committed:
[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

allencal committed:
[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

allencal committed:
[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

allencal committed:
[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

[Traffic Manager] Adding new Realtime User Metrics (RUM) preview feature.

allencal committed:
RUM fixes

Changes for real user metrics

allencal committed:
Merge remote-tracking branch 'upstream/current' into current

# Conflicts:
#	specification/trafficmanager/resource-manager/Microsoft.Network/2017-05-01/trafficmanager.json

allencal committed:
Revert "RUM fixes"

This reverts commit b703b5e7f0f8ae21221387129e49a0114ae68b93.

allencal committed:
RUM fixes

allencal committed:
RUM fixes

allencal committed:
Remove unreferenced metrics key properties / model from 2017-05-01

allencal committed:
Merge branch 'current' of https://github.com/allencal/azure-rest-api-specs into current

allencal committed:
Remove files

lmazuel commented on specification/trafficmanager/resource-manager/Microsoft.Network/2017-09-01-preview/trafficmanageranalytics.json:
```diff
@@ -430,30 +427,32 @@
+      "properties": {
+        "id": {
+          "type" : "string",
+          "description": "Fully qualified resource Id for the resource. Ex - /providers/Microsoft.Network/trafficManagerUserMetricsKeys/default"
```
It seems you removed "default" evrywhere else, is the description still accurate?

allencal replied:
You're right, I think I should change this here. The "/default" is still returned in this description, though it shouldn't be; the call with "/default" on the end fails. I'll remove it here, and also in the API operation.

lmazuel commented on specification/trafficmanager/resource-manager/Microsoft.Network/2017-09-01-preview/trafficmanageranalytics.json:
```diff
@@ -126,14 +126,11 @@
               "$ref": "#/definitions/TrafficManagerUserMetricsKeyModel" 
             }
           },
-          "204": {
```
Does this mean that 204 is impossible, or you want now a 204 to raise an exception?

allencal replied:
The "Get" operation never returns a 204. If the subscription has no RUM key, it returns a 200 and the body returns a blank key ("key": "")

allencal committed:
[TrafficManager] Real User Metrics (RUM) fixes II - remove default from resource ID and operation ID

[TrafficManager] Real User Metrics (RUM) fixes II - remove default from resource ID and operation ID

lmazuel reviewed (approved):
LGTM. Tell me if you sign-off or if you have more to add.

allencal commented:
I do not have more to add. I'm not authorized to merge this pull request. What happens next?

lmazuel commented:
@allencal Next is I merge :)
If your question is "next-next" like SDK/CLI, please send me an email (my github alias is my MS alias)

AutorestCI commented:
No modification for AutorestCI/azure-sdk-for-node

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR1920)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/1597
